### ISSUE TITLE
H5 Gridstack.js completely useless in safari

### ISSUE THREAD

**Issue by kevinlandsberg** [gridstack/gridstack.js#1540]
## Subject of the issue
Gridstack.js completely useless in safari.

## Your environment
* Gridstack.js version: 3.1.2 with HTML5
* Safari, Version 14.0.2 (16610.3.7.1.9) under MacOS Big Sur

## Steps to reproduce
Just visit libary homepage and try to use examples.

## Expected behavior
Tiles should be draggable

## Actual behavior
No dragging ability...

**Comment by adumesny** [comment]
will take me a while to try this on a mac as I'm on windows...

**Comment by kevinlandsberg** [comment]
Yeah,

but it seems to get more importance to support Safari cause Apple is selling a lot of computers atm 👍🏿

**Comment by kevinlandsberg** [comment]
You can try out Browserstack (https://www.browserstack.com/)

**Comment by shmolf** [comment]
JavaScript Console Error
```
TypeError: $('.newWidget').draggable is not a function. (In '$('.newWidget').draggable({
      revert: 'invalid',
      scroll: false,
      appendTo: 'body',
      helper: 'clone'
    })', '$('.newWidget').draggable' is undefined)
    ```

**Comment by adumesny** [comment]
@shmolf sounds like you are mixing html5 and jquery-ui there... I don't think that pertains to the site demos but I have not looked on safari yet.

**Comment by shmolf** [comment]
> @shmolf sounds like you are mixing html5 and jquery-ui there... I don't think that pertains to the site demos but I have not looked on safari yet.

Ack! You're probably right. I'll try changing my assets, and let you know the result.

**Comment by shmolf** [comment]
I was using the following CDN URL
https://cdn.jsdelivr.net/npm/gridstack@3.1.2/dist/gridstack-h5.js

This was referenced by the [JS Fiddle example](https://jsfiddle.net/adumesny/jqhkry7g).

Nothing else is loaded, other than CSS.

**Comment by shmolf** [comment]
We may independently use Jquery UI, but not for Gridstack.

Does Gridstack check if Jquery UI is globally available, then references if so?

**Comment by adumesny** [comment]
please file a separate bug with reproduceable example. And no we don't reference JQ ui if it's available...onlyif our JQ plugin is used.

**Comment by shshen** [comment]
I have same issue in safari, when i drag a grid element.   I'm using latest release build:

[Error] ReferenceError: Can't find variable: DragEvent
	initEvent (gridstack-h5.js:2:58075)
	_dragStart (gridstack-h5.js:2:41983)
	_dragStart

**Comment by adumesny** [comment]
for now you will need to use the jquery version (mouse event, not html5 drag&drop as apparently Safari does't support it event though doc says it does since 3.1 (it failed on latest 12.1.2 for me....)

**Comment by adumesny** [comment]
I spent a lot of time over the break and think I have a fix for Safari (and also better behavior for Chrome). Help or Donations would be appreciated..

**Comment by kevinlandsberg** [comment]
I will definetl donate as soon as it will work on Safari and there will be the engine fiy for swapping grid items instead of pushing them down...

**Comment by kevinlandsberg** [comment]
> I spent a lot of time over the break and think I have a fix for Safari (and also better behavior for Chrome). Help or Donations would be appreciated..

I will definetl donate as soon as it will work on Safari and there will be the engine fiy for swapping grid items instead of pushing them down...#149

**Comment by adumesny** [comment]
fixed in upcoming 3.1.3
